.evergreen/config.yml

```diff
@@ -147,7 +147,8 @@ functions:
     - command: "shell.exec"
       params:
         script: |-
-          ${compile_env|} CONFIGURE_ONLY=ON ./libmongocrypt/.evergreen/build_all.sh
+          env ${compile_env|} CONFIGURE_ONLY=ON CC=clang CXX=clang++ \
+              bash libmongocrypt/.evergreen/build_all.sh
```

Comment:
Why this change?

Reply:
`clang-tidy` will read the compile options from `compile_commands.json` generated by CMake. By default CMake will select GCC as the compiler, and the new warnings include `-Wdiscarded-qualifiers`, which is a GCC-only option. Telling CMake to use Clang here will ensure that it generates a Clang-compatible `compile_commands.json`.

Reply:
Thank you for the explanation. TIL `-Wdiscarded-qualifiers` is GCC-only.